action-server/db/reminder.py

```diff
@@ -23,18 +71,103 @@ class Reminder(Base):
+
+        reminder = Reminder(timezone)
+        for slot_name, attribute_name in SLOT_MAPPING.items():
+            reminder._set_attribute(attribute_name, slot_values.get(slot_name, None))
+
+        return reminder
+
+    # For Unit tests
+    def __eq__(self, other):
+        if not isinstance(other, Reminder):
+            return False
+        if self.id != other.id:
+            return False
+        if self.created_at != other.created_at:
+            return False
+        if self.last_reminded_at != other.last_reminded_at:
+            return False
+        if self.timezone != other.timezone:
+            return False
+        if self.is_canceled != other.is_canceled:
+            return False
+        if self.attributes != other.attributes:
+            return False
+        return True
+
+    def __repr__(self):
+        return ", ".join(
+            [
+                "{key}={value}".format(key=key, value=self.__dict__.get(key))
+                for key in self.__dict__
+            ]
+        )
+
+    def _set_attribute(self, attribute, value):
+        if value is not None:
+            self.attributes[attribute] = value
 
     @property
     def first_name(self):
-        return self.attributes["first_name"]
+        return self.attributes[FIRST_NAME_ATTRIBUTE]
+
+    @first_name.setter
+    def first_name(self, first_name):
+        self._set_attribute(FIRST_NAME_ATTRIBUTE, first_name)
 
     @property
     def phone_number(self):
-        return self.attributes["phone_number"]
+        return self.attributes[PHONE_NUMBER_ATTRIBUTE]
+
+    @phone_number.setter
+    def phone_number(self, phone_number):
+        self._set_attribute(PHONE_NUMBER_ATTRIBUTE, phone_number)
 
     @property
     def language(self):
-        return self.attributes["language"]
+        return self.attributes[LANGUAGE_ATTRIBUTE]
+
+    @language.setter
+    def language(self, language):
+        self._set_attribute(LANGUAGE_ATTRIBUTE, language)
+
+    @property
+    def province(self):
+        return self.attributes[PROVINCE_ATTRIBUTE]
+
+    @province.setter
+    def province(self, province):
+        self._set_attribute(PROVINCE_ATTRIBUTE, province)
+
+    @property
+    def age_over_65(self):
+        return self.attributes[AGE_OVER_65_ATTRIBUTE]
+
+    @age_over_65.setter
+    def age_over_65(self, age_over_65):
+        self._set_attribute(AGE_OVER_65_ATTRIBUTE, age_over_65)
+
+    @property
+    def pre_existing_conditions(self):
+        return self.attributes[PRE_EXISTING_CONDITIONS_ATTRIBUTE]
+
+    @pre_existing_conditions.setter
+    def pre_existing_conditions(self, pre_existing_conditions):
+        self._set_attribute(PRE_EXISTING_CONDITIONS_ATTRIBUTE, pre_existing_conditions)
+
+    @property
+    def has_dialogue(self):
+        return self.attributes[HAS_DIALOGUE_ATTRIBUTE]
+
+    @has_dialogue.setter
+    def has_dialogue(self, has_dialogue):
+        self._set_attribute(HAS_DIALOGUE_ATTRIBUTE, has_dialogue)
```

Comment:
We try to use type annotations elsewhere for both parameters and return types. Can you add these?

Reply:
will do!